### PR TITLE
feat(rpc): impl parity trace_transaction

### PR DESCRIPTION
**Pull request by mattsse**
impl the `trace_transaction`

move some stuff around and add some helpers for better ergonomics

### CODE REVIEW

File: crates/revm/revm-inspectors/src/tracing/mod.rs [hunk 1]
```diff
@@ -60,9 +62,14 @@ impl TracingInspector {
         }
     }
 
-    /// Consumes the Inspector and returns the recorded.
-    pub fn finalize(self) -> CallTraceArena {
-        self.traces
+    /// Consumes the Inspector and returns a [ParityTraceBuilder].
+    pub fn into_parity_builder(self) -> ParityTraceBuilder {
+        ParityTraceBuilder::new(self.traces.arena, self.config)
```
Comment (rkrasiuk) [inline crates/revm/revm-inspectors/src/tracing/mod.rs:67]: are traces within the arena sorted in the order of occurrence?

File: crates/revm/revm-inspectors/src/tracing/builder/geth.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    contract_storage.insert(key.into(), value.into());
+                    log.storage = Some(contract_storage.clone());
+                }
+            }
+            if opts.disable_stack.unwrap_or_default() {
+                log.stack = None;
+            }
+            if !opts.enable_memory.unwrap_or_default() {
+                log.memory = None;
+            }
```
Comment (rkrasiuk) [inline crates/revm/revm-inspectors/src/tracing/builder/geth.rs:52]: by default, the stack is enabled and memory is disabled. that's the intention, right?

**Review by rkrasiuk — COMMENTED** [review]
the new code lgtm, i'll need to refresh existing tho

File: crates/revm/revm-inspectors/src/tracing/mod.rs (hunk 1 above)
Comment (mattsse) [reply crates/revm/revm-inspectors/src/tracing/mod.rs:67]: yes

File: crates/revm/revm-inspectors/src/tracing/builder/geth.rs (hunk 2 above)
Comment (mattsse) [reply crates/revm/revm-inspectors/src/tracing/builder/geth.rs:52]: depends on the target, for parity style we don't need any of them.

**Pull request merged**